google/cloud/aiplatform/initializer.py

```diff
@@ -88,8 +90,14 @@ def init(
             utils.validate_region(location)
             self._location = location
         if experiment:
-            logging.warning("Experiments currently not supported.")
-            self._experiment = experiment
+            metadata.metadata_service.set_experiment(experiment)
+        if run:
+            if not experiment:
```

Comment:
Again as mentioned above, we should allow this per our current design.

Reply:
Same comment as before. I think we can still change the design. We can talk more tmrw

Reply:
Discussed offline. We can get this in, and then un-expose if not needed